mlir/lib/Transforms/Utils/FoldUtils.cpp

```diff
@@ -154,6 +154,7 @@ bool OperationFolder::insertKnownConstant(Operation *op, Attribute constValue) {
                                  !isFolderOwnedConstant(op->getPrevNode())))
     op->moveBefore(&insertBlock->front());
 
+  op->setLoc(erasedFoldedLocation);
```

Comment:
I would think we'd only need to erase the location if we actually move the operation? (that is: the check 3 lines above)

Reply:
oh yes it comes down to two policies:
- Erase location when the folder "owns" it, or
- Erase location when the folder manipulates it (hoist it or deduplicate something else with it)

Since there's only one place where the folder "ingests" an existing op to own it, I thought tracking it early would make the logic simpler to maintain. I don't have a really strong preference though. Let me push a patch on top to erase late and we can choose. The benefit with the latter of course is for constants that are already at the top of a block, they're unaffected.

Reply:
> The benefit with the latter of course is for constants that are already at the top of a block, they're unaffected.

Yes that is what I had in mind: the folder taking "ownership" of a constant, but this one never being moved.